include/fizzy/fizzy.h

```diff
@@ -11,8 +11,98 @@
 extern "C" {
 #endif
 
+/// The opaque data type representing a module.
+typedef struct FizzyModule FizzyModule;
+
+/// The opaque data type representing an instance (instantiated module).
+typedef struct FizzyInstance FizzyInstance;
+
+/// The data type representing numeric values.
+///
+/// i64 member is used to represent values of both i32 and i64 type.
+union FizzyValue
+{
+    uint64_t i64;
+    float f32;
+    double f64;
+};
+
+/// Result of execution of a function.
+typedef struct FizzyExecutionResult
+{
+    /// Whether execution ended with a trap.
+    bool trapped;
+    /// Whether function returned a value. Valid only if trapped equals false.
+    bool has_value;
+    /// Value returned from a function. Valid only if has_value equals true.
+    union FizzyValue value;
+} FizzyExecutionResult;
+
+
+/// Pointer to external function.
+///
+/// @param context      Opaque pointer to execution context.
+/// @param instance     Pointer to module instance.
+/// @param args         Pointer to the argument array. Can be NULL iff args_size equals 0.
+/// @param args_size    Size of the  argument array.
+/// @param depth        Call stack depth.
+typedef FizzyExecutionResult (*FizzyExternalFn)(void* context, FizzyInstance* instance,
+    const union FizzyValue* args, size_t args_size, int depth);
+
+/// External function.
+typedef struct FizzyExternalFunction
+{
+    // TODO function type
+
+    /// Pointer to function.
+    FizzyExternalFn function;
+    /// Opaque pointer to execution context, that will be passed to function.
+    void* context;
+} FizzyExternalFunction;
+
+/// Validate binary module.
 bool fizzy_validate(const uint8_t* wasm_binary, size_t wasm_binary_size);
 
+/// Parse binary module.
+const FizzyModule* fizzy_parse(const uint8_t* wasm_binary, size_t wasm_binary_size);
+
+/// Free resources associated with the module.
```

Comment:
Perhaps mention `module` must be non-null.

Reply:
And the same comments apply to instantiate.

Reply:
This calles `delete`, so is guaranteed to be no-op for null pointer. I'll mention this and add a test.